packages/utils/telemetry-utils/src/test/utils.spec.ts

```diff
@@ -69,3 +70,55 @@ describe("tagData", () => {
 		assert.deepEqual(taggedData, expected);
 	});
 });
+
+describe("Sampling", () => {
+	let events: ITelemetryBaseEvent[] = [];
```

Comment:
for the base logger you can reuse the MockLogger, which already supports tracking of events